### PR TITLE
W-10617943: Fix namespace read while picking up connector error type as seen for SAP S4hana Odata connector

### PR DESCRIPTION
**Pull request by mule-pachanady**
We notice some test failures in  Sap - S4hana Odata Mule 4. 
```
+ Running test: config-null-base-url-test - Cause a CONNECTIVITY Error. Thrown +
+ when the base url is null.                                                   +
java.lang.AssertionError: The error ID thrown does not match the expected one.  expected:<[S4HANA]:CONNECTIVITY> but was:<[MULE]:CONNECTIVITY>
```
This is due to recent changes to `ExceptionHandlerManager -> resolveConnectionErrorType()` 11/2/21, where a namespace name S4 HANA does not match expected type S4HANA

` "S4HANA:CONNECTIVITY"`

### CODE REVIEW

**Comment by mule-pachanady** [comment]
@dabreu I think you worked on related code here, doing this change to fix some regression issue in Sap4 HANA connector. Please review when you get a chance.

**Review by dabreu — APPROVED** [review]

**Review by asanguinetti — CHANGES_REQUESTED** [review]
Please add a test covering this case so we make sure it doesn't break in the future.

**Comment by mule-pachanady** [comment]
@asanguinetti Please review

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/exception/ExceptionHandlerManagerTestCase.java [hunk 1]
```diff
@@ -154,6 +159,28 @@ public void handleConnectionExceptionExtensionWithCustomConnectivityError() {
     assertThat(((ConnectionException) throwable).getErrorType().get().getNamespace(), is(EXTENSION_NAMESPACE));
   }
 
+  @Test
```
Comment (asanguinetti) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/exception/ExceptionHandlerManagerTestCase.java:162]: Please add an Allure annotation relating the new test to the issue that required its creation.

Like this:
`@Issue("W-10617943")`

You can also add a more human friendly description to the test with:
`@Description("Your description here")`

**Comment by mule-pachanady** [comment]
--validate

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/exception/ExceptionHandlerManagerTestCase.java (hunk 1 above)
Comment (mule-pachanady) [reply modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/exception/ExceptionHandlerManagerTestCase.java:161]: Done Thanks

**Review by asanguinetti — APPROVED** [review]

**Pull request merged**